### PR TITLE
feat(behavior_path_planner): add a method to notify the planner manager of a successful lane change (update root lanelet)

### PR DESCRIPTION
**Pull request by danielsanchezaran**
## Description

Create a method for the scene modules to notify the planner manager that a lane change has occurred. By default the method returns false, but its overridden for the lane_change scene_module. This change helps simplify the root lanelet change procedure and makes it so you don't need to modify the planner manager to request a lane change if you add a new behavior path module.

## Tests performed

PSim -> perform lane changes bu setting goal in different lanelet and verified that the root lanelet updates after the LC module is successful. 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM!
Thank you so much for super quick work!!!!!!!!

File: planning/behavior_path_planner_common/include/behavior_path_planner_common/interface/scene_module_interface.hpp [hunk 1]
```diff
@@ -259,6 +259,8 @@ class SceneModuleInterface
     return is_waiting_approval_ || current_state_ == ModuleStatus::WAITING_APPROVAL;
   }
 
+  virtual bool isRootLaneletToBeUpdated() { return false; }
```
Comment (rej55) [inline planning/behavior_path_planner_common/include/behavior_path_planner_common/interface/scene_module_interface.hpp:262]: This method can be `const` ?

Comment (danielsanchezaran) [reply planning/behavior_path_planner_common/include/behavior_path_planner_common/interface/scene_module_interface.hpp]: @rej55 Thanks for the suggestion, I have added it in the latest commit

**Review by rej55 — APPROVED** [review]
LGTM

**Comment by danielsanchezaran** [comment]
@rej55 @kyoichi-sugahara 
Thanks for the reviews, I will also wait for @satoshi-ota and then merge if there is no problem

**Pull request merged**